src/Base/Metadata.py

```diff
@@ -0,0 +1,45 @@
+"""
+This file keeps auxiliary metadata to be used by the Python API stubs.
+"""
+
+
+def metadata(**kwargs):
+    """
+    A decorator to attach metadata to a class.
+    """
+
+    def decorator(cls):
+        cls.__metadata__ = kwargs
```

Comment:
Is this a Python magic attribute, or a FreeCAD?
I would like to point to Python documentation that warns for usage of such attributes - these are reserved for python only. This is a frequent error that may have serious consequences (I cannot remember exactly, but probably there was once such a problem in FreeCAD and the attribute must be refactored in an urgent mode).

> Any use of `__*__` names, in any context, that does not follow explicitly documented use, is subject to breakage without warning.

https://docs.python.org/3/reference/lexical_analysis.html#reserved-classes-of-identifiers


I propose to change this to a normal attribute: `stub_metadata`/`freecad_metadata` /`cpp_metadata`.
This remark also applies to other similar attributes, but I will write it here, only once.

Reply:
> I cannot remember exactly, but probably there was once such a problem in FreeCAD and the attribute must be refactored in an urgent mode

I remember at least the Python-3.11 compatibility issue with  `__getstate__`/`__setstate__`. We have it in the [v1.0 release notes](https://wiki.freecad.org/Release_notes_1.0#Changed_Python_API).